projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -801,6 +802,14 @@ private static String getContainerName(V1Job job) {
                   .ifPresent(s -> jobExecutionStatusBuilder.terminationMessage(s));
             jobExecutionStatusBuilder.terminationReason(jobStatusCondition.getReason());
         }
+
+        Optional<V1ContainerStateTerminated> lastContainerState = getTerminationStatus(job);
```

Comment:
Is there any reason for not putting this into the `if` above, where the `lastTerminatedPodState` is already acquired? This will save an extra call to the K8s API to list all of the job pods.

Reply:
From what I understand in terms of the way kubernetes works, it is possible that at the time the `if` block above is executed, the pod of the data job might no longer exist, which would set the container termination reason as `null`.
If my understanding is incorrect, I am happy to move the logic inside the `if`, as that would indeed save extra calls to the K8s API.

Reply:
The code inside the `if` processes the termination message of the pod, so I believe the pod should still exist at this point.

Reply:
Moved the logic inside the `if` block. Thanks for the review